src/newscale/stages.py

```diff
@@ -1,365 +1,497 @@
+
+    # <05>
+    #def time_step(self, direction: Direction, step_count: int = None,
+    #              step_interval_us: float = None,
+    #              step_duration_us: float = None):
+    #    """Setup a specified number of steps to move through at a specific
+    #    interval."""
+    #    # Note: we cannot accept NEITHER as a direction
+    #    raise NotImplementedError
+
+    # <06>
+    def distance_step(self, direction: Direction, step_size_mm: float = None):
+        """Take a step of size `step_size` in the specified direction.
+        If no step size is specified, the previous step size will be used.
+
+        :param direction: direction to step in.
+        :param step_size_mm: the size of the step. Optional.
+        """
+        # TODO: warn if step size was never specified.
+        step_size_ticks = round(step_size_mm*TICKS_PER_MM)  # 32 bit unsigned.
+        assert step_size_ticks.bit_length() < 32, "Step size exceeds maximum."
+        return self._send(self._get_cmd_str(Cmd.STEP_CLOSED_LOOP,
+                                            direction.value,
+                                            f"{step_size_ticks:08x}"))
+
+    # <06> variant
+    def set_distance_step_size(self, step_size_mm: float):
+        """Specify the step size taken (in mm) in :meth:`distance_step`"""
+        return self.distance_step(Direction.NEITHER, step_size_mm)
+
+    # <07>
+    # Skip this one.
+
+    # <08>
+    def move_to_target(self, setpoint_mm: float = None):
+        """Move to the target absolute setpoint specified in mm.
+
+        :param setpoint_mm:  positive or negative setpoint.
+        """
+        # TODO: convert target setpoint to encoder ticks.
+        if setpoint_mm is None:
+            return self._get_cmd_str(Cmd.MOVE_TO_TARGET)
+        step_size_ticks = round(setpoint_mm*TICKS_PER_MM)
+        assert step_size_ticks.bit_length() < 32, "Step size exceeds maximum."
+        return self._send(self._get_cmd_str(Cmd.MOVE_TO_TARGET,
+                                            f"{step_size_ticks:08x}"))
+
+    # <09>
+    def set_open_loop_speed(self, percent: float):
+        """Set the open loop speed as a percent (0 to 100) of the full scale
+        range.
+
+        :param percent: Speed indicated as a percentage (0.0-100.0)
+        """
+        speed_byte = round(percent/100 * 255)
+        assert 1 < speed_byte < 256, "speed setting out of range."
+        return self._send(self._get_cmd_str(Cmd.OPEN_LOOP_SPEED,
+                                            f"{speed_byte:02x}"))
+
+    # <10>
+    def get_closed_loop_state_and_position(self):
+        """Return a tuple of (state as a dict, position in mm, error in mm)."""
+        # use get_state
+        _, state_int, pos, error = \
+            self._send(self._get_cmd_str(Cmd.CLOSED_LOOP_STATE))
+        return self._parse_state(state_int), pos/TICKS_PER_MM, \
+               error/TICKS_PER_MM
+
+    # <10> variant
+    def get_position(self):
```

Comment:
i don't think this is necessary, we can just call the function above and discard the other return values.

Reply:
Same comment as https://github.com/AllenNeuralDynamics/python-newscale/pull/1#discussion_r1001164529

Generally, I think the point of a good driver is to abstract out nuances like these so that the user doesn't need to read the datasheet to use the device. Happy to chat about this in more detail, but it boils down to a "separation of concerns" design pattern.